x-pack/packages/ml/in_memory_table/hooks/use_table_state.ts

```diff
@@ -5,14 +5,25 @@
  * 2.0.
  */
 
-import { useState } from 'react';
+import { Dispatch, SetStateAction, useState } from 'react';
 import { EuiInMemoryTable, Direction, Pagination } from '@elastic/eui';
 
-export function useTableState<T>(items: T[], initialSortField: string) {
+export interface UseTableState<T> {
```

Comment:
Using this instead of return typeof useTableState in order to pass <T> to onTableChange